Argo/Types/JSON.swift

```diff
@@ -28,6 +28,15 @@ public extension JSON {
   }
 }
 
+extension JSON {
+  func jsonForKey(key: Swift.String) -> JSON {
+    switch self {
+    case let .Object(o): return o[key] ?? .Null
+    default: return .Null
```

Comment:
Does this lose the type mismatch error when an unexpected JSON type is encountered? This will end up complaining that the key was missing (in `guardNull`) rather than saying that an object was expected, right?